### ISSUE TITLE
Consistently use GridapType in the project

### ISSUE THREAD

**Issue by fverdugo** [gridap/Gridap.jl#125]
Now, we are using `GridapType` only in some random places.

1. We need to go through all the `abstract type` and `struct` definitions, and when possible, inherit from `GridapType`.

2. Add this function definition so that `GridapType` objects have an alternative printing style much less verbose than the default in julia.

```julia
function show(io::IO,object::GridapType)
  print(io,"$(nameof(typeof(object)))()")
end
```

**Comment by victorsndvg** [comment]
I already did this, see commit 

New struct and abstract types inheriting from `GridapType` are:
  - NewtonRaphsonCache
  - Evaluation
  - Kernel
  - CartesianDescriptor
  - GridTopology
  - Triangulation
  - NFace

I have not found more than these. Is there any other pending?

**Comment by fverdugo** [comment]
Seems OK!

**Comment by fverdugo** [comment]
Closed via PR #130
